concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -770,6 +1506,29 @@ getAncestorsV2 cptr channel blockType blockHashPtr depth outHash cbk = do
             _ <- enqueueMessages (sender channel) modules
             return (queryResultCode QRSuccess)
 
+getTransactionStatusV2 ::
+    StablePtr Ext.ConsensusRunner ->
+    -- |Transaction hash.
+    Ptr Word8 ->
+    Ptr ReceiverVec ->
+    -- |Callback to output data.
+    FunPtr CopyToVecCallback ->
+    IO Int64
+getTransactionStatusV2 cptr trxHashPtr outVec copierCbk = do
+    Ext.ConsensusRunner mvr <- deRefStablePtr cptr
+    let copier = callCopyToVecCallback copierCbk
+    trxHash <- decodeTransactionHashInput trxHashPtr
+    res <- runMVR (Q.getTransactionStatus trxHash) mvr
+    case res of
+        Nothing -> do
+          return $ queryResultCode QRNotFound
+        Just ts -> case toTransactionStatus ts of
+              Left e -> Exception.throw e -- TODO: Review question: What should be returned if the if it is CEFailed* and not CEInvalid*?
```

Comment:
You could extend `QueryResult` with the error and define an error code for it.
We probably also want to log a warning, since these are internal errors.

Reply:
I think what Emil suggests is the correct approach. Return a special code here, and log an internal server error.
The response to the client is likewise an internal server error and "report bug".

Reply:
I've extended the QueryResult now. I just logged with `hPutStrLn` which probably isn't ideal. Let me know what to use ;)